gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ColorsGenerator.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.generator
+
+import com.android.utils.forEach
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeSpec
+import dev.icerock.gradle.generator.android.AndroidColorsGenerator
+import dev.icerock.gradle.generator.common.CommonColorsGenerator
+import dev.icerock.gradle.generator.ios.IosColorsGenerator
+import org.gradle.api.file.FileTree
+import java.io.File
+import javax.xml.parsers.DocumentBuilderFactory
+
+abstract class ColorsGenerator(
+    private val colorsFileTree: FileTree
+) : MRGenerator.Generator {
+
+    override val resourceClassName: ClassName =
+        ClassName("dev.icerock.moko.resources", "ColorResource")
+    override val mrObjectName: String = "colors"
+
+    protected val singleColorClassName =
+        ClassName("dev.icerock.moko.resources", "ColorResource.Single")
+    protected val themedColorClassName =
+        ClassName("dev.icerock.moko.resources", "ColorResource.Themed")
+
+    override fun generate(resourcesGenerationDir: File, objectBuilder: TypeSpec.Builder): TypeSpec {
+        objectBuilder.addModifiers(*getClassModifiers())
+        extendObjectBody(objectBuilder)
+
+        val colors = parseColors()
+        colors.forEach { colorNode ->
+            val className = if (colorNode.isThemed()) {
+                themedColorClassName
+            } else {
+                singleColorClassName
+            }
+            val property = PropertySpec.builder(colorNode.name, className)
+            property.addModifiers(*getPropertyModifiers())
+            getPropertyInitializer(colorNode)?.let { property.initializer(it) }
+            objectBuilder.addProperty(property.build())
+        }
+
+        generateResources(resourcesGenerationDir, colors)
+
+        return objectBuilder.build()
+    }
+
+    override fun extendObjectBody(classBuilder: TypeSpec.Builder) = Unit
+
+    protected open fun getClassModifiers(): Array<KModifier> = emptyArray()
+    protected open fun getPropertyModifiers(): Array<KModifier> = emptyArray()
+    protected open fun getPropertyInitializer(color: ColorNode): CodeBlock? {
+        val className = if (color.isThemed()) {
+            "Themed(light = Color(0x${argbToRgba(color.lightColor)}), dark = Color(0x${argbToRgba(color.darkColor)}))"
+        } else {
+            "Single(color = Color(0x${argbToRgba(color.singleColor)}))"
```

Comment:
maybe in our xml use just RGBA, not ARGB? it's only in android used ARGB as i know

Reply:
Now all colors parses as RGBA.